src/encoder/Encoder.cs

```diff
@@ -2,36 +2,42 @@
 
 namespace PitchConverter.Encoder
 {
-    public abstract class Encoder
+    public abstract class Encoder : IEncoder
     {
         private protected string _pitchClasses;
+        private protected bool _allowNumbers;
         public int StartOctave { get; set; }
         public bool IncludeRests { get; set; }
         public OutputFormat Format { get; init; }
 
         public Encoder(OutputFormat format)
         {
             _pitchClasses = PitchConstants.PitchClassSets.NonChromaticPitchClasses;
+            _allowNumbers = true;
             StartOctave = 4;
             IncludeRests = true;
             Format = format;
         }
 
-        public abstract List<MusicSymbol> Encode(string input);
+        public virtual List<MusicSymbol> Encode(string input)
```

Comment:
Yeah so I typo'd. It's `virtual`, not `abstract`. The code change is correct here.